src/traces/scattergl/index.js

```diff
@@ -807,7 +854,68 @@ function selectPoints(searchInfo, polygon) {
             scene.unselectBatch[i] = [];
         }
         // we should turn scatter2d into unselected once we have any points selected
-        scene.scatter2d.update(scene.unselectedOptions);
+        if(hasMarkers) {
+            scene.scatter2d.update(scene.unselectedOptions);
+        }
+    }
+
+    // update texts selection
+    if(hasText) {
```

Comment:
We'll have to move this to out of `selectPoints` and into `scene.plot` to make set sure e.g. `Plotly.restyle(gd, 'selected.marker.textfont.color', 'red')` works.

Reply:
Fixed in bdc8f4a1a96ed498f0851b775e55ab974bd36265